server/src/main/java/org/opensearch/wlm/cancellation/QueryGroupTaskCancellationService.java

```diff
@@ -70,15 +85,32 @@ public QueryGroupTaskCancellationService(
         this.deletedQueryGroups = deletedQueryGroups;
     }
 
+    public void setQueryGroupStateMapAccessor(final Function<String, QueryGroupState> queryGroupStateAccessor) {
+        this.queryGroupStateAccessor = queryGroupStateAccessor;
+    }
+
     /**
      * Cancel tasks based on the implemented strategy.
      */
-    public final void cancelTasks(BooleanSupplier isNodeInDuress) {
+    public void cancelTasks(BooleanSupplier isNodeInDuress) {
         queryGroupLevelResourceUsageViews = resourceUsageTrackerService.constructQueryGroupLevelUsageViews();
         // cancel tasks from QueryGroups that are in Enforced mode that are breaching their resource limits
         cancelTasks(ResiliencyMode.ENFORCED);
         // if the node is in duress, cancel tasks accordingly.
         handleNodeDuress(isNodeInDuress);
+
+        updateResourceUsageInQueryGroupState();
+    }
+
+    private void updateResourceUsageInQueryGroupState() {
+        for (Map.Entry<String, QueryGroupLevelResourceUsageView> queryGroupLevelResourceUsageViewEntry : queryGroupLevelResourceUsageViews
+            .entrySet()) {
```

Comment:
What is the purpose of this method? and why are we invoking it after `handlingNodeDuress`? Intuitively, resource usages should be refreshed before handling node duress.

Reply:
The order of this method invocation will not matter until there is a refresh of resource stats at task level. This is just to update the current usage in state of a query group so that we can report it as part of stats